src/app/components/Search/Search.jsx

```diff
@@ -16,38 +16,34 @@ import {
 /**
  * The main container for the top Search section.
  */
-class Search extends React.Component {
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      field: this.props.field,
-      searchKeywords: this.props.searchKeywords,
-    };
-
-    this.inputChange = this.inputChange.bind(this);
-    this.submitSearchRequest = this.submitSearchRequest.bind(this);
-    this.triggerSubmit = this.triggerSubmit.bind(this);
-    this.onFieldChange = this.onFieldChange.bind(this);
-
-    // Build ref for the select element (aka search_scope) selector:
-    this.searchByFieldRef = null;
-    this.setSearchByFieldRef = (element) => {
-      this.searchByFieldRef = element;
-    };
-  }
-
-  componentWillReceiveProps(nextProps) {
-    this.setState(nextProps);
-  }
+function Search (props) {
+  const {
+    field,
+    searchKeywords,
+    selectedFilters,
+    createAPIQuery,
+    updateSearchKeywords,
+    updateField,
+    router
+  } = props;
+
+  // Set component level state based on props
+  const [selectField, setSelectField] = useState(field);
+  const [keywords, setKeywords] = useState(searchKeywords);
+
+  useEffect(() => {
+    // Set component level state on prop changes
+    if (searchKeywords !== keywords) setKeywords(searchKeywords)
+    if (field !== selectField) setSelectField(field)
+  }, [searchKeywords, field])
```

Comment:
This is missing `keywords` and `selectField` in the dependency array. From eslint (and "Error Lens" vscode plugin):
<img width="948" alt="Screen Shot 2023-03-29 at 5 18 14 PM" src="https://user-images.githubusercontent.com/1280564/228670311-3ee2ac8d-8cd2-4a67-9210-08388db819e7.png">

Reply:
Adding those variables to the dependency array causes an issue where the state doesn't properly update, I believe because of a circular dependency.